src/Neo.VM/ReferenceCounterV2.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ReferenceCounterV2.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.VM.Types;
+using System;
+
+namespace Neo.VM
+{
+    /// <summary>
+    /// Used for reference counting of objects in the VM.
+    /// </summary>
+    public sealed class ReferenceCounterV2 : IReferenceCounter
+    {
+        /// <summary>
+        /// Gets the count of references.
+        /// </summary>
+        public int Count { get; private set; }
+
+        /// <summary>
+        /// Adds item to Reference Counter
+        /// </summary>
+        /// <param name="item">The item to add.</param>
+        /// <param name="count">Number of similar entries</param>
+        public void AddStackReference(StackItem item, int count = 1)
+        {
+            Count += count;
```

Comment:
As @roman-khimov suggested, take a look at compound types handling in https://github.com/nspcc-dev/neo-go/blob/b8a65d3c37cfa429b8a5e135422bb1fa78762056/pkg/vm/ref_counter.go#L15.

Reply:
We've got some tests as well: https://github.com/nspcc-dev/neo-go/blob/b8a65d3c37cfa429b8a5e135422bb1fa78762056/pkg/vm/ref_counter_test.go